src/View/Form/ContextInterface.php

```diff
@@ -44,6 +44,15 @@ public function isPrimaryKey($field);
     public function isCreate();
 
     /**
+     * Sets the type of the form the context is created from.
+     *
+     * @param string $requestType The request type of the form the
+     *   context is created from.
+     * @return void
+     */
+    public function setRequestType($requestType);
```

Comment:
That was my first approach, actually. I also don't likr modifying the public interface. However, the request type has not been determined once the context is created and even worse, determining the request type relies somewhat on the context (see the create method of the FormHelper class).